common/src/primitives/id.rs

```diff
@@ -181,10 +210,36 @@ mod tests {
             assert_eq!(h.as_bytes(), bytes);
             assert_eq!(u.to_bytes(), bytes);
         }
-        check("0000000000000000000000000000000000000000000000000000000000000000");
-        check("0000000000000000000000000000000000000000000000000000000000000001");
-        check("000000006a625f06636b8bb6ac7b960a8d03705d1ace08b1a19da3fdcc99ddbd");
-        check("02f0000ff000000004ec466ce4732fe6f1ed1cddc2ed4b328fff5224276e3f6f");
-        check("000000000000000000059fa50103b9683e51e5aba83b8a34c9b98ce67d66136c");
+        SAMPLE_HASHES.iter().cloned().for_each(check)
+    }
+
+    #[test]
+    fn h256_json() {
+        fn check(hex: &'static str) {
+            use serde_json::Value;
+            let hash: H256 = hex.parse().unwrap();
+            // hash should serialize into its hex string
+            serde_test::assert_tokens(&hash, &[serde_test::Token::Str(hex)]);
+            assert_eq!(serde_json::to_value(hash).ok(), Some(Value::String(format!("{:x}", hash))));
+        }
+        SAMPLE_HASHES.iter().cloned().for_each(check)
+    }
+
+    #[test]
+    fn id_json() {
+        fn check(hex: &'static str) {
+            use serde_test::Token::{Str, Struct, StructEnd};
+            use serde_json::Value;
+            let id: Id<()> = Id::new(&hex.parse().unwrap());
+            // An id serializes into {"id": "<hex_hash>"}
```

Comment:
I would suggest not making this elaborate and just making it map to a string of the hash. This can have bad consequences when requesting things by id from rpc. For example, if we request the block index of a block by hash, the input from the user should be `{"id": "<hex_hash>"}`, which is some unnecessary pain. Just make it `"<hex_hash>"`.

Reply:
I actually wanted to ask about this. Will change that.